src/FixedRateIrm.sol

```diff
@@ -0,0 +1,50 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.8.19;
```

Comment:
Why not 0.8.21 for instance?

Reply:
always this push0 thing... we can change to 0.8.21 with evm_version=paris yes though

Reply:
Maybe some chains have adopted push0 by now ? Or we can assume that at the moment where Blue will be deployed they will have adpoted push0

Reply:
why not even 0.8.24 actually

Reply:
We used the Fixed Rate IRM Iterface (IFixedRateIrm.sol), to integrate with morpho
We created upper and lower bounds to mitigate the risk of extreme rates on the lower and upper bounds.

`
    // This should set at once when the contract is initially deployed
    uint256 public upperBound = 100;
    uint256 public lowerBound = 0;

    /* SETTER */

    /// @inheritdoc IFixedRateIrm
    function setBorrowRate(Id id, uint256 newBorrowRate) external {
        // require(_borrowRate[id] == 0, ErrorsLib.RATE_ALREADY_SET);
        require(newBorrowRate != 0, ErrorsLib.RATE_ZERO);
        
        // checks to see if the rate is within the lower and upper bound. 
        require(upperBound < newBorrowRate && newBorrowRate  > lowerBound, ErrorsLib.NOT_IN_BOUNDS);
        _borrowRate[id] = newBorrowRate;

        emit SetBorrowRate(id, newBorrowRate);
    }
    `

Reply:
> Maybe some chains have adopted push0 by now ? Or we can assume that at the moment where Blue will be deployed they will have adpoted push0

FYI https://www.evmdiff.com/features?name=PUSH0&kind=opcode

Note that the website only supports Ethereum, Arb and OP.